packages/next/src/build/webpack/plugins/next-types-plugin.ts

```diff
@@ -44,11 +44,11 @@ function createTypeGuardFile(
   }
 ) {
   return `// File: ${fullPath}
-import * as entry from '${relativePath}'
+import * as entry from '${relativePath}.js'
```

Comment:
Is it safe to always import `.js` here even if the user could have e.g. `.ts`?

Reply:
Yes, `.js` is the correct extension for importing a `.ts` file in TypeScript ESM with fully-specified import paths:

- https://github.com/microsoft/TypeScript/issues/41887#issuecomment-741968855

Also the reason that the `resolve.extensionAlias` config was added to webpack:

https://github.com/webpack/webpack/issues/13252#issuecomment-1171080020

Reply:
Thanks!